src/d3d12.rs

```diff
@@ -0,0 +1,2285 @@
+    ) -> ::HRESULT,
+    fn CreateHeap(
+        &mut self, pDesc: *const ::D3D12_HEAP_DESC, riid: ::REFGUID, ppvHeap: *mut *mut ::c_void
+    ) -> ::HRESULT,
+    fn CreatePlacedResource(
+        &mut self, pHeap: *mut ::ID3D12Heap, HeapOffset: ::UINT64,
+        pDesc: *const ::D3D12_RESOURCE_DESC, InitialState: ::D3D12_RESOURCE_STATES,
+        pOptimizedClearValue: *const ::D3D12_CLEAR_VALUE, riid: ::REFGUID,
+        ppvResource: *mut *mut ::c_void
+    ) -> ::HRESULT,
+    fn CreateReservedResource(
+        &mut self, pDesc: *const ::D3D12_RESOURCE_DESC, InitialState: ::D3D12_RESOURCE_STATES,
+        pOptimizedClearValue: *const ::D3D12_CLEAR_VALUE, riid: ::REFGUID,
+        ppvResource: *mut *mut ::c_void
+    ) -> ::HRESULT,
+    fn CreateSharedHandle(
+        &mut self, pObject: *mut ::ID3D12DeviceChild, pAttributes: *const ::SECURITY_ATTRIBUTES,
+        Access: ::DWORD, Name: ::LPCWSTR, pHandle: *mut ::HANDLE
+    ) -> ::HRESULT,
+    fn OpenSharedHandle(
+        &mut self, NTHandle: ::HANDLE, riid: ::REFGUID, ppvObj: *mut *mut ::c_void
+    ) -> ::HRESULT,
+    fn OpenSharedHandleByName(
+        &mut self, Name: ::LPCWSTR, Access: ::DWORD, pNTHandle: *mut ::HANDLE
+    ) -> ::HRESULT,
+    fn MakeResident(
+        &mut self, NumObjects: ::UINT, ppObjects: *mut *mut ::ID3D12Pageable
+    ) -> ::HRESULT,
+    fn Evict(
+        &mut self, NumObjects: ::UINT, ppObjects: *mut *mut ::ID3D12Pageable
+    ) -> ::HRESULT,
+    fn CreateFence(
+        &mut self, InitialValue: ::UINT64, Flags: ::D3D12_FENCE_FLAGS, riid: ::REFGUID,
+        ppFence: *mut *mut ::c_void
+    ) -> ::HRESULT,
+    fn GetDeviceRemovedReason(&mut self) -> ::HRESULT,
+    fn GetCopyableFootprints(
+        &mut self, pResourceDesc: *const ::D3D12_RESOURCE_DESC, FirstSubresource: ::UINT,
+        NumSubresources: ::UINT, BaseOffset: ::UINT64,
+        pLayouts: *mut ::D3D12_PLACED_SUBRESOURCE_FOOTPRINT, pNumRows: *mut ::UINT,
+        pRowSizeInBytes: *mut ::UINT64, pTotalBytes: *mut ::UINT64
+    ) -> (),
+    fn CreateQueryHeap(
+        &mut self, pDesc: *const ::D3D12_QUERY_HEAP_DESC, riid: ::REFGUID,
+        ppvHeap: *mut *mut ::c_void
+    ) -> ::HRESULT,
+    fn SetStablePowerState(&mut self, Enable: ::BOOL) -> ::HRESULT,
+    fn CreateCommandSignature(
+        &mut self, pDesc: *const ::D3D12_COMMAND_SIGNATURE_DESC,
+        pRootSignature: *mut ::ID3D12RootSignature, riid: ::REFGUID,
+        ppvCommandSignature: *mut *mut ::c_void
+    ) -> ::HRESULT,
+    fn GetResourceTiling(
+        &mut self, pTiledResource: *mut ::ID3D12Resource, pNumTilesForEntireResource: *mut ::UINT,
+        pPackedMipDesc: *mut ::D3D12_PACKED_MIP_INFO,
+        pStandardTileShapeForNonPackedMips: *mut ::D3D12_TILE_SHAPE,
+        pNumSubresourceTilings: *mut ::UINT, FirstSubresourceTilingToGet: ::UINT,
+        pSubresourceTilingsForNonPackedMips: *mut ::D3D12_SUBRESOURCE_TILING
+    ) -> (),
+    fn GetAdapterLuid(&mut self, __ret_val: *mut ::LUID) -> *mut ::LUID
```

Comment:
Nope, it's actually correct since `stdcall` actually differs depending on whether it is a method or a function. In particular in certain cases when returning an aggregate, it actually takes an extra parameter which is a pointer to where to store the return value, and it also returns that pointer. This was carefully tested.